### PR TITLE
Reduce repetitive code of Frontend Roadmap Component

### LINKED ISSUE #291: Improve UI and Maintainability of Frontend Roadmap Component
The current implementation of the Frontend component lacks structure and maintainability. The component can be improved by enhancing its responsiveness, readability, and overall styling. This involves organizing the data, **reducing repetitive code**, and applying better styling practices.

### Proposed Changes:
1. Enhance Responsiveness and  Improve Styling
2. **Simplify JSX:**  Reduce repetitive code by using array mapping for course elements.

### Detailed Steps:
1. Create an array of course objects containing name and link properties.
2. Map through the array to generate course elements dynamically.
3. Use conditional rendering to manage the visibility of arrow images based on screen size.
4. Apply consistent styling to the container and course elements.

### PR DESCRIPTION
**Pull request by NegiSushant**
**Description**
This pull request improves the UI and maintainability of the frontend Roadmap component. The changes include:

- Organizing course data in an array and mapping through it for rendering.
- Enhancing responsiveness using Tailwind CSS utilities.
- Improving styling by adding a background color and adjusting padding.
- Simplifying JSX by reducing repetitive code.

These changes make the component easier to manage, more responsive, and visually appealing.

This PR fixes #291

### CODE REVIEW

File: src/components/frontend.jsx [hunk 1]
```diff
@@ -1,63 +1,56 @@
...
-            </div>
+  return (
+    <div className="bg-gray-800 min-h-screen py-20">
+      <h1 className="w-full mb-24 text-white px-4 tracking-tight text-center text-5xl">
+        Complete Frontend Roadmap
+      </h1>
 
-        </div>
-    );
+      <div className="flex flex-wrap justify-center gap-4 mx-4 md:mx-20">
```
Comment (Mr-Sunglasses) [inline src/components/frontend.jsx:24]: You can use `grid-cols-7` which allows for better control over the placement of the courses and arrows.

File: src/components/frontend.jsx [hunk 2]
```diff
@@ -1,63 +1,56 @@
...
+        {courses.map((course, index) => (
+          <div key={index}>
```
Comment (Mr-Sunglasses) [inline src/components/frontend.jsx:26]: You can also use[ React.Fragment](https://react.dev/reference/react/Fragment) to group each course with its corresponding arrow which can help ensure that the arrow is placed in the next grid column.

**Review by Mr-Sunglasses — CHANGES_REQUESTED** [review]

**Comment by NegiSushant** [comment]
hey @Mr-Sunglasses ! I review my code and the result show me without any change is: 
image

File: src/components/frontend.jsx (hunk 2 above)
Comment (NegiSushant) [reply src/components/frontend.jsx:26]: got it! fixed.

**Comment by Mr-Sunglasses** [comment]
Yeah @NegiSushant It looks good, please push those changes :-)

**Review by Mr-Sunglasses — APPROVED** [review]
Thanks @NegiSushant

**Pull request merged**